drivers/input/input_chsc6x.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Nicolas Goualard <nicolas.goualard@sfr.fr>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT seeed_chsc6x
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/input/input.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+
+struct chsc6x_config {
+	struct i2c_dt_spec i2c;
+	const struct gpio_dt_spec int_gpio;
+};
+
+struct chsc6x_data {
+	const struct device *dev;
+	struct k_work work;
+	struct gpio_callback int_gpio_cb;
+};
+
+struct chsc6x_output {
+	uint8_t points;
+	uint16_t x;
+	uint16_t y;
+} __packed;
+
+LOG_MODULE_REGISTER(chsc6x, CONFIG_INPUT_LOG_LEVEL);
+
+static int chsc6x_process(const struct device *dev)
+{
+	uint16_t row, col;
+	bool is_pressed;
+
+	struct chsc6x_output output;
+	const struct chsc6x_config *cfg = dev->config;
+
+	int res = i2c_burst_read_dt(&cfg->i2c, 0, (uint8_t *)&output, sizeof(output));
+
+	if (res < 0) {
+		LOG_ERR("Could not read data %i", res);
+		return -ENODATA;
+	}
+
+	is_pressed = output.points & 0xff;
+	col = sys_be16_to_cpu(output.x) & 0x00ff;
+	row = sys_be16_to_cpu(output.y) & 0x00ff;
```

Comment:
these looks off:
- is points the number of touches? if that's the case the logic should probably be "points > 1" rather than the mask
- the row/col what's the point of doing a 16bit be/le swap just mask out the output to the lowest 8 bits? at this point you may as well just transfer an array of 8 bytes and pick the right ones

Reply:
- The chip does not support multiple touches, so is_pressed is either 0 or 1.
- I took inspiration from some other drivers, but when I first started testing it it seemed that the MSB of output.x and/or output.y was not always 0, so I put a mask, but it's simpler to just use the correct bytes, yes. I also did that for is_pressed in my changes.